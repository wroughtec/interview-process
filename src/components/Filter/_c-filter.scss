.c-filter__input {
  width: 100%;
  height: 3rem;
  padding: .5rem .75rem;

  border: 1px solid var(--inputBorder);

  transition: border-color .15s ease-in-out;

  &::placeholder {
    opacity: 1;
  }
}
