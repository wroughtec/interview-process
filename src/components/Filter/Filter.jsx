// @flow
import React from 'react';
import './_c-filter.scss';

type Props = {
  handleChange: Function,
  filter: string
};

export const Filter = ({ handleChange, filter }: Props) => (
  <label className="c-filter" htmlFor="filter">
    <span className="sr-only">Search:</span>
    <input
      className="c-filter__input"
      type="search"
      id="filter"
      placeholder="Search..."
      onChange={handleChange}
      value={filter}
    />
  </label>
);
