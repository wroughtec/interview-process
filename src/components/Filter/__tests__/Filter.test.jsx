import React from 'react';
import { shallow } from 'enzyme';
import { Filter } from 'components/Filter/Filter';

describe('Filter', () => {
  let component, props;
  beforeEach(() => {
    props = {
      handleChange: jest.fn(),
      filter: 'test'
    };

    component = shallow(<Filter {...props} />);
  });
  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('on change function should have been called', () => {
    component.find('input').simulate('change', { currentTarget: { value: 'change' } });

    expect(props.handleChange).toHaveBeenCalled();
  });
});
