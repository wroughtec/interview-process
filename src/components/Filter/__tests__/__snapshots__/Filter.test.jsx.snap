// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Filter
    filter="test"
    handleChange={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="sr-only"
        >
          Search:
        </span>,
        <input
          className="c-filter__input"
          id="filter"
          onChange={[MockFunction]}
          placeholder="Search..."
          type="search"
          value="test"
        />,
      ],
      "className": "c-filter",
      "htmlFor": "filter",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Search:",
          "className": "sr-only",
        },
        "ref": null,
        "rendered": "Search:",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "c-filter__input",
          "id": "filter",
          "onChange": [MockFunction],
          "placeholder": "Search...",
          "type": "search",
          "value": "test",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "label",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="sr-only"
          >
            Search:
          </span>,
          <input
            className="c-filter__input"
            id="filter"
            onChange={[MockFunction]}
            placeholder="Search..."
            type="search"
            value="test"
          />,
        ],
        "className": "c-filter",
        "htmlFor": "filter",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search:",
            "className": "sr-only",
          },
          "ref": null,
          "rendered": "Search:",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "c-filter__input",
            "id": "filter",
            "onChange": [MockFunction],
            "placeholder": "Search...",
            "type": "search",
            "value": "test",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "label",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
